### ISSUE TITLE
False negative from `no-unnecessary-act`

### ISSUE THREAD

**Issue by Belco90** [testing-library/eslint-plugin-testing-library#686]
### Discussed in https://github.com/testing-library/eslint-plugin-testing-library/discussions/685

<div type='discussions-op-text'>

<sup>Originally posted by **douglasjunior** October 28, 2022</sup>
In one test case we called a function within ACT that ESLint indicates is a testing-library function, but is not.

![image](https://user-images.githubusercontent.com/1512341/198692772-eca3d026-f671-46d9-b6cd-399317f01ab1.png)

And `dismissCallback` need to be wrapped with `act` because it dispatches a state update to close the AlertModal.

```tsx
  it('should not auto close modal with autoDismiss=false', async () => {
    const mockOnDismiss = jest.fn();

    let dismissCallback: undefined | (() => void);

    const tree = render(
      <AlertModal
        visible
        title="Title"
        description="Lorem ipsum is placeholder text commonly used in the graphic"
        primaryButtonName="Primary Action"
        secondaryButtonName="Secondary Action"
        autoDismiss={false}
        onPressSecondary={event => {
          dismissCallback = event.dismiss;
        }}
        onDismiss={mockOnDismiss}
      />,
    );

    const buttonInstance = tree.getByTestId('alert-modal-secondary-button');
    fireEvent.press(buttonInstance);

    expect(dismissCallback).toBeDefined();

    expect(mockOnDismiss).not.toBeCalled();

    act(() => {
      dismissCallback?.();
    });

    expect(mockOnDismiss).toHaveBeenCalled();
```

---

This has been confirmed as a false positive. Removing the optional chaining from `dismissCallback` call fixes the issue:

![imagen](https://user-images.githubusercontent.com/2677072/199014619-90cf6a79-8bd3-451b-97bb-b9dbebac1a4f.png)

But wrapping it in an if statement gets the problem back:

![imagen](https://user-images.githubusercontent.com/2677072/199014685-75c4a575-073e-46d3-b43d-803c26fadf65.png)

This seems related to the fact that `CallExpression` is not a direct child of the function body, so we are not detecting this correctly.
